### PR TITLE
search macro improvements (see issue #157)

### PR DESCRIPTION
**Pull request by TheGeorge**
`?EXIST` and `?NOT_EXIST` do not require the `#{gen=>...}` thingy anymore and there is a `?FORALL_TARGETED` that works like the exists and is not specialized on a search strategy (which should be specified in the proper options)

### CODE REVIEW

File: src/proper_statem.erl [hunk 1]
```diff
@@ -218,7 +218,7 @@
 
 -module(proper_statem).
 
--export([commands/1, commands/2, parallel_commands/1, parallel_commands/2,
+-export([commands/1, commands/2, commands/4, parallel_commands/1, parallel_commands/2,
```
Comment (fenollp) [inline src/proper_statem.erl]: Where is this used by the "targeted" part of PropEr?

Comment (TheGeorge) [reply src/proper_statem.erl]: I should read my diffs more carefully (or maybe don't make pull requests that late). Thanks for pointing it out.

**Comment by fenollp** [comment]
Note: Travis CI's `make all` probably needs a `rebar3 xref` equivalent. Would have caught that typo :)

**Comment by TheGeorge** [comment]
So after adding the test case, I discovered a bug, where shrinking did not work, when TPBT was used together with a constructed '?LET' generator. This PR should fix this bug (I also added a test to check for this).

**Comment by TheGeorge** [comment]
I also played around with the test macros and they should now not timeout anymore.

**Comment by kostis** [comment]
> I also played around with the test macros and they should now not timeout anymore.

This is nice, of course, but do we have any idea why the timeout was happening?  Using 1000 for a timeout value seems/is a bit excessive -- at least compared with the other timeout values in the test suite.

**Comment by TheGeorge** [comment]
Since they seem to work now I set them to 1 min which should be plenty of time. From my understanding, the timeout does not behave well when it is specified for multiple tests as it was before.

File: test/exists_tests.erl [hunk 2]
```diff
@@ -386,16 +399,51 @@ shrink3_test() ->
...
 prop_shrink4() ->
-  ?NOT_EXISTS(I, #{gen => integer(0, 20)},
+  ?NOT_EXISTS(I, integer(0, 20),
               begin
                 ?MINIMIZE(I),
                 I < 10
               end).
+
+matching_type() ->
+        ?LET(I, integer(), I).
```
Comment (kostis) [inline test/exists_tests.erl]: Indent me properly, please.

**Comment by ferd** [comment]
nice, had not seen this PR go through before! Thanks for the improvements!

**Pull request merged**